coderd/searchquery/search_test.go

```diff
@@ -163,6 +222,11 @@ func TestSearchWorkspace(t *testing.T) {
 			Query:                 `foo:bar`,
 			ExpectedErrorContains: `"foo" is not a valid query param`,
 		},
+		{
+			Name:                  "ParamExtraColons",
+			Query:                 "param:foo:value",
+			ExpectedErrorContains: "can only contain 1 ':'",
```

Comment:
double-check: will this break parameter values like `ghcr.io/coder/coder-preview:main` (docker image with colon)

Reply:
It will work! I added a test case to show this, You have to quote the value:
```
param:"image=ghcr.io/coder/coder-preview:main"
```

Reply:
Btw the reason this works and the `=` does not is because the split on the colon is done by the `searchTerms` function here: https://github.com/coder/coder/blob/767287492b60d75ff1cf480ac304718a75b7bdaa/coderd/searchquery/search.go#L162-L162

It finds the `key:value` syntax. The equal sign however is done by a value parser. So from the search term code POV, `key:value=value`, `value=value` is a single term that is passed to the parser.

To fix this, we would need to make the parser smarter. Then you can argue should it support quotes? Escape characters? Etc... For now, I think it's not worth it. This is primarily for backstage atm.